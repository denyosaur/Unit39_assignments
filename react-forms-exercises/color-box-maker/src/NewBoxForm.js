import React, { useState } from "react";

const NewBoxForm = ({ createBox }) => {
    const INITIAL_STATE = {
        height: "",
        width: "",
        color: "#000000"
    }
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleChange = (evt) => {
        evt.preventDefault();
        const { name, value } = evt.target;
        setFormData({
            ...formData,
            [name]: value
        });
    };

    const handleSubmit = (evt) => {
        evt.preventDefault();
        createBox(formData.height, formData.width, formData.color);
        setFormData(INITIAL_STATE);
    };

    return <form onSubmit={handleSubmit}>
        <label htmlFor="height">Height:</label>
        <input name="height" type="number" placeholder="Enter Height Number" onChange={handleChange} value={formData.height} />

        <label htmlFor="width">Width: </label>
        <input name="width" type="number" placeholder="Enter Width Number" onChange={handleChange} value={formData.width} />

        <label htmlFor="color">Color: </label>
        <input name="color" type="color" placeholder="Enter Color" onChange={handleChange} value={formData.color} />

        <button>Create</button>
    </form>


};

export default NewBoxForm;