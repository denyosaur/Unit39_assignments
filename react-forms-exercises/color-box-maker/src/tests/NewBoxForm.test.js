import { fireEvent, render, screen } from '@testing-library/react';
import NewBoxForm from '../NewBoxForm';

//smoke test for Box
test('renders Box without crashing', () => {
    render(<NewBoxForm />);
});

//snapshot test for Box
test('should match snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

