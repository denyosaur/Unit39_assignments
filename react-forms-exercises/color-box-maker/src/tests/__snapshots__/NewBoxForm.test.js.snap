// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="height"
    >
      Height:
    </label>
    <input
      name="height"
      placeholder="Enter Height Number"
      type="number"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      name="width"
      placeholder="Enter Width Number"
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      name="color"
      placeholder="Enter Color"
      type="color"
      value="#000000"
    />
    <button>
      Create
    </button>
  </form>
</DocumentFragment>
`;
